src/main/java/com/example/bookshop/service/implementation/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,83 @@
+package com.example.bookshop.service.implementation;
+
+import com.example.bookshop.dto.cartitem.CartItemRequestDto;
+import com.example.bookshop.dto.cartitem.ShoppingCartDto;
+import com.example.bookshop.exception.EntityNotFoundException;
+import com.example.bookshop.mapper.ShoppingCartMapper;
+import com.example.bookshop.model.CartItem;
+import com.example.bookshop.model.ShoppingCart;
+import com.example.bookshop.model.User;
+import com.example.bookshop.repository.BookRepository;
+import com.example.bookshop.repository.CartItemRepository;
+import com.example.bookshop.repository.ShoppingCartRepository;
+import com.example.bookshop.repository.UserRepository;
+import com.example.bookshop.service.ShoppingCartService;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+@Transactional
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private ShoppingCartRepository shoppingCartRepository;
+    private ShoppingCartMapper shoppingCartMapper;
+    private BookRepository bookRepository;
+    private UserRepository userRepository;
+    private CartItemRepository cartItemRepository;
+
```

Comment:
let's create shoppingCart when user registers. Create method in `ShoppingCartService` and use it `UserService`